server/routes.go

```diff
@@ -1487,17 +1489,48 @@ func (s *Server) ChatHandler(c *gin.Context) {
 				},
 			}
 
+			if len(req.Tools) > 0 {
```

Comment:
Do we need this check? Ultimately it lets us flush a little sooner, however will it be that much faster than sending back one chunk?

Reply:
Actually I think this check is not needed anymore since we have checking later - although by doing this check we'd skip over `parseToolCalls` which would be a slight performance gain. Since this is the main flow I think I'll leave this in